### ISSUE TITLE
CI is crashing with memory issues

### ISSUE THREAD

**Issue by TimvdLippe** [mockito/mockito#2478]
On a bunch of recent PRs, the following error is thrown in the GitHub actions:

```
Exception: java.lang.OutOfMemoryError thrown from the UncaughtExceptionHandler in thread "Test worker"
```

Examples: https://github.com/mockito/mockito/runs/4264231004?check_suite_focus=true and https://github.com/mockito/mockito/runs/4264151319?check_suite_focus=true

It is unclear what has changed, as we weren't running into this issue before. Additionally, my build completes just fine locally.

Can a recent change to GitHub Actions maybe have lowered the amount of memory we can use?

**Comment by TimvdLippe** [comment]
I inspected a Gradle build scan and it appears that it is crashing on `large_stack_trace_invocations_should_not_starve_memory`. That is not unsurprising, but we should still investigate as to why that is happening. I am going to disable that test and see if CI is green.

**Comment by TimvdLippe** [comment]
Disabling the test appears to have worked. I am rerunning CI on all other PRs to get these in. We should of course look into re-enabling these tests, but it isn't clear to me where to start.

**Comment by TimvdLippe** [comment]
This should be fixed with #2723
